pycqed/instrument_drivers/physical_instruments/ZurichInstruments/UHFQuantumController.py

```diff
@@ -199,24 +214,89 @@ def __init__(self,
         # Used for extra DIO output to CC for debugging
         self._diocws = None
 
+        # Holds the DIO calibration delay
+        self._dio_calibration_delay = 0
+
+        # Define parameters that should not be part of the snapshot
+        self._params_to_exclude = set(['features_code', 'system_fwlog', 'system_fwlogenable'])
+
         # Our base class includes all the functionality needed to initialize the parameters
         # of the object. Those parameters are read from instrument-specific JSON files stored
         # in the zi_parameter_files folder.
-        super().__init__(name=name, device=device, interface=interface, server=server, port=port, **kw)
+        super().__init__(name=name, device=device, interface=interface,
+                         server=server, port=port, num_codewords=2**nr_integration_channels,
+                         **kw)
+
+        # Disable disfunctional parameters from snapshot
+        self._params_to_exclude = set(['features_code', 'system_fwlog', 'system_fwlogenable'])
 
         # Set default waveform length to 20 ns at 1.8 GSa/s
         self._default_waveform_length = 32
 
+        # Mask used for detecting codeword activity during DIO calibration
+        self._dio_calibration_mask = None
+
         t1 = time.time()
-        print('Initialized UHFQC', self.devname, 'in %.2fs' % (t1-t0))
+        log.info(f'{self.devname}: Initialized UHFQC in {t1 - t0}s')
 
     ##########################################################################
-    # Private methods
+    # Overriding Qcodes InstrumentBase methods
+    ##########################################################################
+
+    def snapshot_base(self, update: bool=False,
+                      params_to_skip_update =None,
+                      params_to_exclude = None ):
```

Comment:
- Not mentioned in the doctoring. 
- Also it is not clear to me, how the effect of this parameter differs from params_to_skip.